azure/functions/_abc.py

```diff
@@ -421,3 +424,88 @@ class OrchestrationContext(abc.ABC):
     @abc.abstractmethod
     def body(self) -> str:
         pass
+
+class RequestStatus(Enum):
+     Failed = auto()
+     TokenInvalid = auto()
+     Successful = auto()
+
+class IEventResponse(abc.ABC):
+    def __init__(self,
+                 schema : typing.Optional[str],
+                 body: typing.Optional[str],
+                 jsonBody: typing.Optional[str]):
+                 self.schema= schema
+                 self.body=body
+                 self.jsonBody=jsonBody
+
+
+    def invalidate():
+        pass
+    
+    @staticmethod
+    def createInstance(type : type, schema : str, body : str):
+        response =IEventResponse(type())
+        response.Schema = schema
+        response.Body = body
+        return response
+
+class IActionable(abc.ABC):
+
+        abc.abstractmethod    
+        def invalidateActions():
+            pass
+    
+class IEventAction(abc.ABC):
+    def __init__(self,
+                actionType: str):
+                self.actionType=actionType
+    
+    abc.abstractmethod
+    def buildActionBody():
+        pass
+
+class ITokenIssuanceAction(IEventAction):
```

Comment:
This must be in the _authenticationevents, as it's only related to the on token issuance start event